// @nanometrics/pegasus-harvest-lib/nodejs-api/models/constants.js
"use strict";
exports.__esModule = true;
var UpperLimit;
(function (UpperLimit) {
    UpperLimit[UpperLimit["timestamp"] = Number.MAX_SAFE_INTEGER] = "timestamp";
    UpperLimit[UpperLimit["sequence_number"] = 4294967294] = "sequence_number"; /* max(sequence_number) = max(uint32_t) = 2^32 - 1,
      but C++ library reserves this value for PSF_INVALID_SEQUENCE_INDEX -> use 2^32 - 2
    */
})(UpperLimit = exports.UpperLimit || (exports.UpperLimit = {}));
var HarvestSubDirectory;
(function (HarvestSubDirectory) {
    HarvestSubDirectory["seismicData"] = "";
    HarvestSubDirectory["soh"] = "soh";
    HarvestSubDirectory["timingSoh"] = "timing_soh";
    HarvestSubDirectory["forensicLogs"] = "logs";
})(HarvestSubDirectory = exports.HarvestSubDirectory || (exports.HarvestSubDirectory = {}));
exports.OutputPatterns = {
    SDS: '${Y}/${N}/${S}/${C}.D/${N}.${S}.${L}.${C}.D.${Y}.${J}',
    BUD: '${N}/${S}/${S}.${N}.${L}.${C}.${Y}.${J}'
};
/**
 * See https://docs.google.com/spreadsheets/d/1s3q3Ywi3yDhK-BG_c-baTRWCgXVpr_03KemhfIAerns/edit#gid=0 for more details.
 * These values should be kept in sync.
 */
exports.BinCodes = {
    land: [
        "ASM19053-131",
        "ASM19053-140",
        "ASM19053V2-131",
        "ASM19053V2-140",
        "ASM19458-140",
        "ASM19458V2-140" /* Pegasus Polar V2 */
    ],
    obs: [
        "ASM19361-01", /* Pegasus OBS V1 */ ,
        "ASM19361-02",
        "ASM19481-01",
        "ASM19481-02" /* Pegasus OBS V2 (NMX-Supplied Seascan) */
    ]
};
