// @nanometrics/pegasus-harvest-lib/nodejs-api/models/pdb-constants.js
"use strict";
exports.__esModule = true;
var PDBVolumeID;
(function (PDBVolumeID) {
    PDBVolumeID[PDBVolumeID["VOLUME_NONE"] = 0] = "VOLUME_NONE";
    PDBVolumeID[PDBVolumeID["VOLUME_SENSOR_TIME_SERIES"] = 1] = "VOLUME_SENSOR_TIME_SERIES";
    PDBVolumeID[PDBVolumeID["VOLUME_HEALTH_TIME_SERIES"] = 2] = "VOLUME_HEALTH_TIME_SERIES";
    PDBVolumeID[PDBVolumeID["VOLUME_CLOCK_STATUS"] = 3] = "VOLUME_CLOCK_STATUS";
    PDBVolumeID[PDBVolumeID["VOLUME_HEALTH_DIGEST"] = 4] = "VOLUME_HEALTH_DIGEST";
    PDBVolumeID[PDBVolumeID["VOLUME_OPERATION_LOG"] = 5] = "VOLUME_OPERATION_LOG";
    PDBVolumeID[PDBVolumeID["VOLUME_HARVEST_LOG"] = 6] = "VOLUME_HARVEST_LOG";
    PDBVolumeID[PDBVolumeID["VOLUME_FORENSIC_LOG"] = 7] = "VOLUME_FORENSIC_LOG";
    PDBVolumeID[PDBVolumeID["VOLUME_HEALTH_STATE"] = 8] = "VOLUME_HEALTH_STATE";
    PDBVolumeID[PDBVolumeID["VOLUME_HEALTH_PDF"] = 9] = "VOLUME_HEALTH_PDF";
    PDBVolumeID[PDBVolumeID["VOLUME_SEISMIC_PDF"] = 10] = "VOLUME_SEISMIC_PDF";
    PDBVolumeID[PDBVolumeID["VOLUMES_COUNT"] = 11] = "VOLUMES_COUNT";
})(PDBVolumeID = exports.PDBVolumeID || (exports.PDBVolumeID = {}));
var PDBMessageID;
(function (PDBMessageID) {
    PDBMessageID[PDBMessageID["MSG_NONE"] = 0] = "MSG_NONE";
    PDBMessageID[PDBMessageID["MSG_SENSOR_INPUT"] = 1] = "MSG_SENSOR_INPUT";
    PDBMessageID[PDBMessageID["MSG_HEALTH_INPUT"] = 2] = "MSG_HEALTH_INPUT";
    PDBMessageID[PDBMessageID["MSG_CLOCK_STATUS"] = 3] = "MSG_CLOCK_STATUS";
    PDBMessageID[PDBMessageID["MSG_HEALTH_DIGEST"] = 4] = "MSG_HEALTH_DIGEST";
    PDBMessageID[PDBMessageID["MSG_DIGITIZER_START"] = 5] = "MSG_DIGITIZER_START";
    PDBMessageID[PDBMessageID["MSG_SMART_SENSOR"] = 6] = "MSG_SMART_SENSOR";
    PDBMessageID[PDBMessageID["MSG_GNSS_CONTROL"] = 7] = "MSG_GNSS_CONTROL";
    PDBMessageID[PDBMessageID["MSG_BLE_CONTROL"] = 8] = "MSG_BLE_CONTROL";
    PDBMessageID[PDBMessageID["MSG_OPAQUE_DATA"] = 9] = "MSG_OPAQUE_DATA";
    PDBMessageID[PDBMessageID["MSG_HARVEST_LOG"] = 10] = "MSG_HARVEST_LOG";
    PDBMessageID[PDBMessageID["MSG_FORENSIC_LOG"] = 11] = "MSG_FORENSIC_LOG";
    PDBMessageID[PDBMessageID["MSG_OBS_SYNC"] = 12] = "MSG_OBS_SYNC";
    PDBMessageID[PDBMessageID["MSG_SOH_DATA"] = 13] = "MSG_SOH_DATA";
})(PDBMessageID = exports.PDBMessageID || (exports.PDBMessageID = {}));
//! PDB message types (packed messageID and versionID)
function MAKE_PDB_MSG_TYPE(messageID, versionID) { return (messageID << 16) | versionID; }
var PDBMessageType;
(function (PDBMessageType) {
    PDBMessageType[PDBMessageType["MSG_SENSOR_INPUT_V0"] = MAKE_PDB_MSG_TYPE(PDBMessageID.MSG_SENSOR_INPUT, 0x0000)] = "MSG_SENSOR_INPUT_V0";
    PDBMessageType[PDBMessageType["MSG_HEALTH_INPUT_V0"] = MAKE_PDB_MSG_TYPE(PDBMessageID.MSG_HEALTH_INPUT, 0x0000)] = "MSG_HEALTH_INPUT_V0";
    PDBMessageType[PDBMessageType["MSG_CLOCK_STATUS_V0"] = MAKE_PDB_MSG_TYPE(PDBMessageID.MSG_CLOCK_STATUS, 0x0000)] = "MSG_CLOCK_STATUS_V0";
    PDBMessageType[PDBMessageType["MSG_HEALTH_DIGEST_V0"] = MAKE_PDB_MSG_TYPE(PDBMessageID.MSG_HEALTH_DIGEST, 0x0000)] = "MSG_HEALTH_DIGEST_V0";
    PDBMessageType[PDBMessageType["MSG_DIGITIZER_START_V0"] = MAKE_PDB_MSG_TYPE(PDBMessageID.MSG_DIGITIZER_START, 0x0000)] = "MSG_DIGITIZER_START_V0";
    PDBMessageType[PDBMessageType["MSG_SMART_SENSOR_V0"] = MAKE_PDB_MSG_TYPE(PDBMessageID.MSG_SMART_SENSOR, 0x0000)] = "MSG_SMART_SENSOR_V0";
    PDBMessageType[PDBMessageType["MSG_GNSS_CONTROL_V0"] = MAKE_PDB_MSG_TYPE(PDBMessageID.MSG_GNSS_CONTROL, 0x0000)] = "MSG_GNSS_CONTROL_V0";
    PDBMessageType[PDBMessageType["MSG_BLE_CONTROL_V0"] = MAKE_PDB_MSG_TYPE(PDBMessageID.MSG_BLE_CONTROL, 0x0000)] = "MSG_BLE_CONTROL_V0";
    PDBMessageType[PDBMessageType["MSG_OPAQUE_DATA_V0"] = MAKE_PDB_MSG_TYPE(PDBMessageID.MSG_OPAQUE_DATA, 0x0000)] = "MSG_OPAQUE_DATA_V0";
    PDBMessageType[PDBMessageType["MSG_HARVEST_LOG_V0"] = MAKE_PDB_MSG_TYPE(PDBMessageID.MSG_HARVEST_LOG, 0x0000)] = "MSG_HARVEST_LOG_V0";
    PDBMessageType[PDBMessageType["MSG_FORENSIC_LOG_V0"] = MAKE_PDB_MSG_TYPE(PDBMessageID.MSG_FORENSIC_LOG, 0x0000)] = "MSG_FORENSIC_LOG_V0";
    PDBMessageType[PDBMessageType["MSG_OBS_SYNC_V0"] = MAKE_PDB_MSG_TYPE(PDBMessageID.MSG_OBS_SYNC, 0x0000)] = "MSG_OBS_SYNC_V0";
    PDBMessageType[PDBMessageType["MSG_SOH_DATA_V0"] = MAKE_PDB_MSG_TYPE(PDBMessageID.MSG_SOH_DATA, 0x0000)] = "MSG_SOH_DATA_V0";
})(PDBMessageType = exports.PDBMessageType || (exports.PDBMessageType = {}));
//! PDB message types (packed messageID and versionID)
function BIT(bit) { return (1 << bit); }
var PDBSohChannelID;
(function (PDBSohChannelID) {
    PDBSohChannelID[PDBSohChannelID["ATU"] = BIT(0)] = "ATU";
    PDBSohChannelID[PDBSohChannelID["VAN"] = BIT(1)] = "VAN";
    PDBSohChannelID[PDBSohChannelID["VCE"] = BIT(2)] = "VCE";
    PDBSohChannelID[PDBSohChannelID["VCO"] = BIT(3)] = "VCO";
    PDBSohChannelID[PDBSohChannelID["VCQ"] = BIT(4)] = "VCQ";
    PDBSohChannelID[PDBSohChannelID["VDP"] = BIT(5)] = "VDP";
    PDBSohChannelID[PDBSohChannelID["VDT"] = BIT(6)] = "VDT";
    PDBSohChannelID[PDBSohChannelID["VE1"] = BIT(7)] = "VE1";
    PDBSohChannelID[PDBSohChannelID["VE2"] = BIT(8)] = "VE2";
    PDBSohChannelID[PDBSohChannelID["VEI"] = BIT(9)] = "VEI";
    PDBSohChannelID[PDBSohChannelID["VEL"] = BIT(10)] = "VEL";
    PDBSohChannelID[PDBSohChannelID["VGL"] = BIT(11)] = "VGL";
    PDBSohChannelID[PDBSohChannelID["VGT"] = BIT(12)] = "VGT";
    PDBSohChannelID[PDBSohChannelID["VLA"] = BIT(13)] = "VLA";
    PDBSohChannelID[PDBSohChannelID["VLO"] = BIT(14)] = "VLO";
    PDBSohChannelID[PDBSohChannelID["VM1"] = BIT(15)] = "VM1";
    PDBSohChannelID[PDBSohChannelID["VM2"] = BIT(16)] = "VM2";
    PDBSohChannelID[PDBSohChannelID["VM3"] = BIT(17)] = "VM3";
    PDBSohChannelID[PDBSohChannelID["VM4"] = BIT(18)] = "VM4";
    PDBSohChannelID[PDBSohChannelID["VPH"] = BIT(19)] = "VPH";
    PDBSohChannelID[PDBSohChannelID["VPL"] = BIT(20)] = "VPL";
    PDBSohChannelID[PDBSohChannelID["VPO"] = BIT(21)] = "VPO";
    PDBSohChannelID[PDBSohChannelID["VPP"] = BIT(22)] = "VPP";
    PDBSohChannelID[PDBSohChannelID["VPT"] = BIT(23)] = "VPT";
    PDBSohChannelID[PDBSohChannelID["VPV"] = BIT(24)] = "VPV";
    PDBSohChannelID[PDBSohChannelID["VNS"] = BIT(25)] = "VNS";
    PDBSohChannelID[PDBSohChannelID["VST"] = BIT(26)] = "VST";
    PDBSohChannelID[PDBSohChannelID["ALL_CHANNELS"] = 134217727] = "ALL_CHANNELS";
    PDBSohChannelID[PDBSohChannelID["NO_PUBLIC_AVAILABILITY_CHANNELS"] = PDBSohChannelID.VGL | PDBSohChannelID.VPT] = "NO_PUBLIC_AVAILABILITY_CHANNELS";
    PDBSohChannelID[PDBSohChannelID["PUBLIC_CHANNELS_OBS"] = PDBSohChannelID.ALL_CHANNELS & ~(PDBSohChannelID.NO_PUBLIC_AVAILABILITY_CHANNELS)] = "PUBLIC_CHANNELS_OBS";
    PDBSohChannelID[PDBSohChannelID["PUBLIC_CHANNELS_LAND"] = PDBSohChannelID.ALL_CHANNELS & ~(PDBSohChannelID.NO_PUBLIC_AVAILABILITY_CHANNELS | PDBSohChannelID.VPO)] = "PUBLIC_CHANNELS_LAND"; // public availability channels (LAND)
})(PDBSohChannelID = exports.PDBSohChannelID || (exports.PDBSohChannelID = {}));
