// @nanometrics/pegasus-harvest-lib/nodejs-api/models/pdb-messages.js
"use strict";
exports.__esModule = true;
var MsgDigitizerStartReason;
(function (MsgDigitizerStartReason) {
    MsgDigitizerStartReason[MsgDigitizerStartReason["START_REASON_POWER_ON"] = 0] = "START_REASON_POWER_ON";
    MsgDigitizerStartReason[MsgDigitizerStartReason["START_REASON_HARVEST"] = 1] = "START_REASON_HARVEST";
    MsgDigitizerStartReason[MsgDigitizerStartReason["START_REASON_FIRMWARE"] = 2] = "START_REASON_FIRMWARE";
    MsgDigitizerStartReason[MsgDigitizerStartReason["START_REASON_ERROR_RECOVER"] = 3] = "START_REASON_ERROR_RECOVER";
    MsgDigitizerStartReason[MsgDigitizerStartReason["START_REASON_CONFIG_UPDATE"] = 4] = "START_REASON_CONFIG_UPDATE";
})(MsgDigitizerStartReason = exports.MsgDigitizerStartReason || (exports.MsgDigitizerStartReason = {}));
var HardwareID;
(function (HardwareID) {
    HardwareID[HardwareID["HARDWARE_ID_UNKNOWN"] = 0] = "HARDWARE_ID_UNKNOWN";
    HardwareID[HardwareID["HARDWARE_ID_STM32H7"] = 1] = "HARDWARE_ID_STM32H7";
})(HardwareID = exports.HardwareID || (exports.HardwareID = {}));
